iceoryx_posh/source/popo/building_blocks/chunk_queue_popper.cpp

```diff
@@ -113,27 +113,41 @@ void ChunkQueuePopper::clear() noexcept
     } while (true);
 }
 
-/// @deprecated #25
-cxx::expected<ChunkQueueError>
-ChunkQueuePopper::attachSemaphore(const mepoo::SharedPointer<posix::Semaphore>& semaphore) noexcept
+bool ChunkQueuePopper::attachConditionVariableSignaler(ConditionVariableData* conditionVariableDataPtr) noexcept
 {
-    if (isSemaphoreAttached())
+    /// @todo Add lock guard here, use smart_lock
+    if (isConditionVariableSignalerAttached())
     {
-        LogWarn() << "Semaphore already set. Attaching the semaphore a second time will be ignored!";
-        return cxx::error<ChunkQueueError>(ChunkQueueError::SEMAPHORE_ALREADY_SET);
+        LogWarn() << "Condition variable signaler already set. Attaching a second time will be ignored!";
+        return false;
     }
     else
     {
-        getMembers()->m_semaphore = semaphore;
-        getMembers()->m_semaphoreAttached.store(true, std::memory_order_release);
-        return cxx::success<void>();
+        getMembers()->m_conditionVariableDataPtr = conditionVariableDataPtr;
+        getMembers()->m_conditionVariableAttached.store(true, std::memory_order_release);
```

Comment:
Do we really need the bool flag? nullptr or != nullptr has the same information and it will be protected with a lock when introducing the locking policy as next step

Reply:
Done